euth/dashboard/forms.py

```diff
@@ -263,7 +270,7 @@ def _create_phase(self, phase, commit, module):
                 Offlinephase.objects.get_or_create(phase=new_phase)
 
     def save(self, commit=True):
-
+        self.clean()
```

Comment:
Why is it necessary to call clean here? Should this be done by the view before calling save()?